docs/docs/included-server-middleware.md

```diff
@@ -0,0 +1,203 @@
+# Included Server Middleware
+
+Http4s includes some middleware out of the box in the `org.http4s.server.middleware`
+package. Some of it is documented in its own page:
+
+* [Authentication]
+* Cross Origin Resource Sharing ([CORS])
+* Response Compression ([GZip])
+* [HSTS]
+
+We'll describe and provide examples for the remaining middleware, but first we set up our service:
+```scala mdoc:silent
+import cats.effect._
+import cats.syntax.all._
+import org.typelevel.ci._
+import org.http4s._
+import org.http4s.dsl.io._
+import org.http4s.implicits._
+import cats.effect.unsafe.IORuntime
+implicit val runtime: IORuntime = cats.effect.unsafe.IORuntime.global
+val service = HttpRoutes.of[IO] {
+  case GET -> Root / "bad" => BadRequest()
+  case POST -> Root / "post" => Ok()
+  case _ => Ok()
+}
+
+val goodRequest = Request[IO](Method.GET, uri"/")
+val badRequest = Request[IO](Method.GET, uri"/bad")
+val postRequest = Request[IO](Method.POST, uri"/post")
+```
+
+## Headers
+
+### Caching
+This middleware adds response headers so that clients know how to cache a response. It performs no caching.
+Below is one example of usage, see [Caching] for more methods.
+
+```scala mdoc:silent
+import org.http4s.server.middleware.Caching
+import scala.concurrent.duration._
+
+val cacheService = Caching.cache(
+  3.hours,
+  isPublic = Left(CacheDirective.public),
+  methodToSetOn = _ == Method.GET,
+  statusToSetOn = _.isSuccess,
+  service
+).orNotFound
+
+```
+```scala mdoc
+cacheService(goodRequest).unsafeRunSync().headers
+cacheService(badRequest).unsafeRunSync().headers
+cacheService(postRequest).unsafeRunSync().headers
+```
```

Comment:
i've used this trick to show only the interesting output, if there's a way to do that in a single block i'd love to know